pkg/azurefile/controllerserver.go

```diff
@@ -213,3 +237,35 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
 	return nil, status.Error(codes.Unimplemented, "ControllerExpandVolume is not yet implemented")
 }
+
+// File share names can contain only lowercase letters, numbers, and hyphens,
+// and must begin and end with a letter or a number,
+// and must be from 3 through 63 characters long.
+// The name cannot contain two consecutive hyphens.
+func setFileShareName(fileShareName, reqName string) (string, error) {
```

Comment:
rename to `getValidFileShareName` and should not return error, we do the filtering, and if original vol name is still invalid, generte a old UUID style share name, and also write a unit test for this func